Packs/OPSWAT-Filescan/Integrations/OPSWATFilescan/OPSWATFilescan.py

```diff
@@ -0,0 +1,390 @@
+            ok_codes=([200]),
+            url_suffix="/users/me",
+        )
+        return request_result
+
+    def post_sample(self, args: Dict[str, Any]) -> Dict[str, Any]:
+        data = {}
+
+        if description := args.get("description"):
+            data["description"] = description
+        if tags := args.get("tags"):
+            data["tags"] = tags
+        if password := args.get("password"):
+            data["password"] = password
+        if is_private := args.get("is_private"):
+            data["is_private"] = is_private
+
+        if url := args.get("url"):
+            data["url"] = url
+
+            return self._http_request(
+                method="POST", url_suffix="/scan/url", ok_codes=([200]), data=data
+            )
+
+        elif entry_id := args.get("entry_id"):
+            try:
+                file_entry = demisto.getFilePath(entry_id)
+            except Exception as e:
+                raise DemistoException(
+                    f'Failed to find file entry with id:"{entry_id}". got error: {e}'
+                )
+
+            with open(file_entry["path"], "rb") as file:
+                return self._http_request(
+                    method="POST",
+                    url_suffix="/scan/file",
+                    ok_codes=([200]),
+                    data=data,
+                    files={"file": file},
+                )
+        else:
+            raise DemistoException("No file or URL was provided.")
+
+    def get_scan_result(self, flow_id: str) -> Dict[str, Any]:
+
+        filters = [
+            "filter=general",
+            "filter=finalVerdict",
+            "filter=allTags",
+            "filter=overallState",
+            "filter=taskReference",
+            "filter=subtaskReferences",
+            "filter=allSignalGroups",
+        ]
+
+        postfix = "&".join(filters)
+
+        url_suffix = f"/scan/{flow_id}/report?{postfix}"
+
+        response = self._http_request(
+            method="GET",
+            ok_codes=([200]),
+            url_suffix=url_suffix,
+        )
+
+        return response
+
+    def get_search_query(self, query_str: str) -> Dict[str, Any]:
+        return self._http_request(
+            method="GET",
+            ok_codes=([200]),
+            params={"query": query_str, "page_size": 10, "page": 1},
+            url_suffix="/reports/search",
+        )
+
+
+""" HELPER FUNCTIONS """
+
+
+def build_one_reputation_result(report: Dict[str, Any]):
+    score = Common.DBotScore.NONE
+
+    final_verdict = report.get("finalVerdict", {})
+    verdict = final_verdict.get("verdict")
+    if verdict.upper() == "BENIGN" or verdict.upper() == "INFORMATIONAL":
+        score = Common.DBotScore.GOOD
+    elif verdict.upper() == "MALICIOUS" or verdict.upper() == "LIKELY_MALICIOUS":
+        score = Common.DBotScore.BAD
+    elif verdict.upper() == "SUSPICIOUS":
+        score = Common.DBotScore.SUSPICIOUS
+    else:
+        score = Common.DBotScore.NONE
```

Comment:
```suggestion

```

not needed if you set `score = Common.DBotScore.NONE` at line 110.

Reply:
Else part removed. Thanks the comment!